controllers/migrate.go

```diff
@@ -0,0 +1,74 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// migratePeerPodsLimit moves the PeerPodConfig "Limit" value to peer-pods-cm
+func (r *KataConfigOpenShiftReconciler) migratePeerPodsLimit() error {
+	peerPodConfig := &unstructured.Unstructured{}
+	peerPodConfig.SetAPIVersion("confidentialcontainers.org/v1alpha1")
+	peerPodConfig.SetKind("PeerPodConfig")
+
+	err := r.Client.Get(context.TODO(), client.ObjectKey{
+		Name:      peerpodConfigCrdName,
+		Namespace: OperatorNamespace,
+	}, peerPodConfig)
+
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.Log.Info("No PeerPodConfig found, skipping migration")
+			return nil
+		}
+		return err
+	}
+
+	limitValue, found, _ := unstructured.NestedString(peerPodConfig.Object, "spec", "limit")
+	// TODO: Not sure if we need to set the default value here, or just skip the
+	// upgrade.
+	if !found {
+		limitValue = "10"
+	}
+
+	configMap := &corev1.ConfigMap{}
+	err = r.Client.Get(context.TODO(), client.ObjectKey{
+		Name:      "peer-pods-cm",
+		Namespace: OperatorNamespace,
+	}, configMap)
+
+	if err != nil {
+		return err
+	}
+
+	configMap.Data["PEERPODS_LIMIT_PER_NODE"] = limitValue
+	err = r.Client.Update(context.TODO(), configMap)
+	if err != nil {
+		return err
+	}
+
+	// TODO: Are we going to delete the object, or just flag it as migrated?
```

Comment:
I think we should delete it. This object was created automatically by the previous version of our operator, so my expectation is that we should dispose of it if we don't need it anymore.

Reply:
My concern here is that I'm not sure if we are migrating all data from that object and we might loose data.

Reply:
Deleting should be fine. If you look at the spec other than the limit, rest of the params (secret, configmap etc) are already provided to the CAA pod - https://github.com/confidential-containers/cloud-api-adaptor/blob/v0.9.0/src/peerpodconfig-ctrl/api/v1alpha1/peerpodconfig_types.go#L24